tests/manual-test-cases/Group5-Functional-Tests/5-4-High-Availability.robot

```diff
@@ -232,15 +232,19 @@ Test
     \     Should Be Equal As Integers  ${rc}  0
     \     Should Be Equal  ${output}  running
     \     ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} rm -f ${c}
-    \     Should Be Equal As Integers  ${rc}  0
+    \     Run Keyword If  ${rc} != 0  Should Not Contain  ${output}  ESX host is temporarily disconnected.
+    \     ...  ELSE  Should Be Equal As Integer  ${rc}  0
```

Comment:
From what I understand, this `ELSE` statement isn't completely accurate either - it's a no-op since the above `Run Keyword If` condition checks that `rc != 0`. Even with the ELSE statement, it's possible for the test to pass without `rm -f` succeeding.

Since the original test expects `rm -f` to succeed (1), I don't think we should change it to pass without that condition. We can consider the following, assuming that we'd like to preserve (1).

If `rm -f` returns a non-zero code, the test _should fail_ - the most we should do is log the error string. This can be done with:

```
${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} rm -f ${c}
Log To Console  ${output}
Should Be Equal As Integers  ${rc}  0
```

This is why the intent of https://github.com/vmware/vic/issues/6580 seemed confusing to me - if the test expects the operation to fail, we can check for the error string. However, that's not the case - the test expects `rm -f` to succeed.

Reply:
yeah, good point.